cpp/include/cudf/lists/detail/scatter.cuh

```diff
@@ -613,6 +614,94 @@ struct list_child_constructor {
                                    stream.value(),
                                    mr);
   }
+
+  /**
+   * @brief (Recursively) constructs child columns that are structs.
+   */
+  template <typename T>
+  std::enable_if_t<std::is_same<T, struct_view>::value, std::unique_ptr<column>> operator()(
+    rmm::device_uvector<unbound_list_view> const& list_vector,
+    cudf::column_view const& list_offsets,
+    cudf::lists_column_view const& source_lists_column_view,
+    cudf::lists_column_view const& target_lists_column_view,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr) const
+  {
+    auto source_column_device_view =
+      column_device_view::create(source_lists_column_view.parent(), stream);
+    auto target_column_device_view =
+      column_device_view::create(target_lists_column_view.parent(), stream);
+    auto source_lists = cudf::detail::lists_column_device_view(*source_column_device_view);
+    auto target_lists = cudf::detail::lists_column_device_view(*target_column_device_view);
+
+    auto source_structs = source_lists_column_view.child();
+    auto target_structs = target_lists_column_view.child();
+
+    auto num_child_rows = get_num_child_rows(list_offsets, stream);
+
+    auto num_struct_members =
+      std::distance(source_structs.child_begin(), source_structs.child_end());
+    std::vector<std::unique_ptr<column>> child_columns;
+    child_columns.reserve(num_struct_members);
+
+    auto project_member_as_list = [stream, mr](column_view const& structs_member,
+                                               cudf::size_type const& structs_list_num_rows,
+                                               column_view const& structs_list_offsets,
+                                               rmm::device_buffer const& structs_list_nullmask,
+                                               cudf::size_type const& structs_list_null_count) {
+      return cudf::make_lists_column(structs_list_num_rows,
+                                     std::make_unique<column>(structs_list_offsets, stream, mr),
+                                     std::make_unique<column>(structs_member, stream, mr),
+                                     structs_list_null_count,
+                                     rmm::device_buffer(structs_list_nullmask),
+                                     stream.value(),
+                                     mr);
+    };
+
+    for (int i{0}; i < num_struct_members; ++i) {
```

Comment:
This definitely looks like a two-input `transform` to me...  If the captures bother you, just use `[&]`?

Reply:
> If the captures bother you, just use [&]?

I've been stung on another PR for overly broad captures. In this case, it does seem justified. Much cleaner now, IMHO. Thanks, @harrism.